ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1703,6 +1703,127 @@ class EclCmdStatus : public EclCmdCommon
+                continue;
+            }
+            if (iter.matchOption(optProblemDesc, ECLOPT_PROBLEM_DESC))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            eclCmdOptionMatchIndicator ind = EclCmdCommon::matchCommandLineOption(iter, true);
+            if (ind != EclCmdOptionMatch)
+                return ind;
+        }
+        return retVal;
+    }
+    virtual bool finalizeOptions(IProperties *globals)
+    {
+        if (!EclCmdCommon::finalizeOptions(globals))
+            return false;
+        return true;
+    }
+    virtual int processCMD()
+    {
+        Owned<IClientWsWorkunits> client = createCmdClient(WsWorkunits, *this);
+
+        Owned<IClientWUCreateZAPInfoRequest> zapReq = client->createWUCreateZAPInfoRequest();
+        zapReq->setWuid(optWuid.get());
+        zapReq->setIncludeThorSlaveLog(optIncThorSlave);
+        zapReq->setProblemDescription(optProblemDesc.get());
+
+        Owned<IClientWUCreateZAPInfoResponse> zapResp = client->WUCreateZAPInfo(zapReq);
+        int ret = outputMultiExceptionsEx(zapResp->getExceptions());
+        if (ret == 0)
+        {
+            StringBuffer filePath(optPath);
+            const char *p = filePath.str();
+            p = (p + strlen(p) - 1);
+            if (!streq(p, PATHSEPSTR))
+                filePath.append(PATHSEPSTR);
+
+            filePath.append(zapResp->getZAPFileName());
+
+            const MemoryBuffer & zapFileData = zapResp->getThefile();
+
+            Owned<IFile> zapFile = createIFile(filePath.str());
+            Owned<IFileIO> zapFileIo = zapFile->open(IFOcreate);
+            Owned<IFileIOStream> zapFileStream = createBufferedIOStream(zapFileIo);
+
+            size32_t written = zapFileStream->write(zapFileData.length(), (const void *)zapFileData.toByteArray());
+
+            if (written != zapFileData.length())
+                throw MakeStringException(-1, "truncated write to file %s, ZAP file creation cancelled.\n", filePath.str());
+
+            fprintf(stdout, "ZAP file written into %s.\n",filePath.str());
+        }
+
+        return 0;
+    }
+    virtual void usage()
+    {
+        fputs("\nUsage:\n"
+            "\n"
+            "Create and store ZAP file of the given workunit.\n"
+            "\n"
+            "ecl zapgen <WUID> -path <zap_file_path>\n"
+            "\n"
+            "   WUID                    workunit ID\n"
+            "   path                    path to store ZAP file\n"
+            " Options:\n"
+            "   --inc-thor-slave-logs   include Thor salve(s) log into the ZAP file\n"
+            "   --description <text>    problem description string\n",
+            stdout);
+        EclCmdCommon::usage();
+    }
+private:
+    StringAttr         optWuid;
+    StringAttr         optPath;
+    bool               optIncThorSlave = false;
+    StringAttr         optProblemDesc;
+
+    EclObjectParameter optObj;
```

Comment:
Besides being initialized in the constructor is optObj used anywhere?  This class "EclObjectParameter" is meant for ECL based functions that might use a file, or a query, or a workunit, etc.

Reply:
Removed